tck/tck-runtime/src/main/java/servlet/tck/api/jakarta_servlet_http/httpservletrequest/HttpServletRequestTests.java

```diff
@@ -685,23 +685,16 @@ public void doHeadTest() throws Exception {
             if (responseBodyHead != null && !responseBodyHead.isEmpty()) {
                 throw new Exception("HEAD request included a response body");
             }
-            // Validate the response headers names
-//            Set<String> headersToMatch = new HashSet<>();
-//            Map<String, List<String>> headersGet = responseGet.getResponseHeaders();
-//            for (Map.Entry<String, List<String>> header : headersGet.entrySet()) {
-//                switch(header.getKey().toLowerCase(Locale.ENGLISH)) {
-//                    case "date":
-//                        // Ignore date header as it will change between requests
-//                        break;
-//                    default:
-//                        headersToMatch.add(header.getKey());
-//                }
-//            }
 
             List<Header> headersGet = responseGet.getResponseHeaders();
             Set<String> headersToMatch = headersGet.stream()
                     .map(Header::getName)
+                    // Filter date header as it will change between requests
                     .filter(name -> !name.toLowerCase(Locale.ENGLISH).equals("date"))
+                    // Filter headers for which a value is determined during content generation
+                    // RFC 9110, section 9.32
+                    .filter(name -> !name.toLowerCase(Locale.ENGLISH).equals("content-length"))
```

Comment:
you will need to ignore `content-length` later as well or this will fail as 

```
diff --git a/tck/tck-runtime/src/main/java/servlet/tck/api/jakarta_servlet_http/httpservletrequest/HttpServletRequestTests.java b/tck/tck-runtime/src/main/java/servlet/tck/api/jakarta_servlet_http/httpservletrequest/HttpServletRequestTests.java
index 9ca4590..3db5fdc 100644
--- a/tck/tck-runtime/src/main/java/servlet/tck/api/jakarta_servlet_http/httpservletrequest/HttpServletRequestTests.java
+++ b/tck/tck-runtime/src/main/java/servlet/tck/api/jakarta_servlet_http/httpservletrequest/HttpServletRequestTests.java
@@ -699,8 +699,8 @@ public class HttpServletRequestTests extends HttpRequestClient {
 
             List<Header> headersHead = responseHead.getResponseHeaders();
             for (Header header : headersHead) {
-                if (header.getName().equalsIgnoreCase("date")) {
-                    // Skip date header
+                if (header.getName().equalsIgnoreCase("date") || header.getName().equalsIgnoreCase("content-length")) {
+                    // Skip date and content-length header
                     continue;
                 }
                 if (!headersToMatch.remove(header.getName())) {

```

Reply:
Thanks for spotting that. I've pushed an updated fix.